app/models/course_subject.rb

```diff
@@ -8,5 +12,27 @@ class CourseSubject < ApplicationRecord
   has_many :user_course_subjects, dependent: :destroy
   has_many :user_courses, through: :user_course_subjects
 
-  validates :duration, presence: true
+  def current_duration
+    return duration unless duration.nil?
+
+    subject.duration
+  end
+
+  private
+  def update_user_course
+    course = Course.find_by id: course_id
+    subjects = course.subjects
+    course.users.each do |user|
+      subjects.each do |subject|
```

Comment:
Vòng lặp kiểu này thừa nhé. Đáng lẽ với mỗi subject được create thì chỉ chạy một lần ứng với subject vừa tạo thôi chứ nhỉ, đây lại cứ create một cái mới lại chạy lại toàn bộ các subject của course. Nói chung xem lại vòng lặp này.

Reply:
Xem lại nhé. Tôi lấy ví dụ đơn giản là thêm một môn mới, ở đây sẽ check lại toàn bộ các môn cũ, thay vì vậy, ta chỉ quan tâm đến môn được create và gọi tới after_create ở đây thôi. Bỏ vòng lặp về subjects này đi.